samples: matter: Add ICD LIT functionality (#16392)
nrfconnect/sdk-nrf: samples/matter/light_switch/README.rst

    :start-after: matter_light_bulb_sample_configuration_file_types_start
    :end-before: matter_light_bulb_sample_configuration_file_types_end

.. _matter_light_switch_snippets:

Snippets
========

.. |snippet| replace:: :makevar:`light_switch_SNIPPET`

.. include:: /includes/sample_snippets.txt

The following snippet is available:

* ``lit_icd`` - Enables experimental LIT ICD support.

  .. note::
    This snippet requires custom ZAP configuration, located in :file:`snippets/lit_icd` directory.
    When using this snippet, remember to configure and generate the correct ZAP files.

>     To do it run the ```west zap-generate``` and choose the :file:`snippets/lit_icd/light_switch.zap` file.

Reviewer: But this is already done and the proper files are pulled in by the snippet itself, right? So it would be valid only in case someone wants to modify the data model in a custom way.

Author: yes, I think I can modify it this way:
```suggestion
    When using this snippet, remember to configure and generate the correct ZAP files if custom data model is needed.
    To do it run the ```west zap-generate``` and choose the :file:`snippets/lit_icd/light_switch.zap` file.
```

>     This snippet requires custom ZAP configuration, located in :file:`snippets/lit_icd` directory.
>     When using this snippet, remember to configure and generate the correct ZAP files.
>     To do it run the ```west zap-generate``` and choose the :file:`snippets/lit_icd/light_switch.zap` file.

Reviewer 2: ```suggestion
    This snippet requires a custom ZAP configuration, located in the :file:`snippets/lit_icd` directory.
    Ensure to configure and generate the correct ZAP files if a custom data model is needed.
    To do this, run the ``west zap-generate`` command and choose the :file:`snippets/lit_icd/light_switch.zap` file.
```
This suggestions includes the one from the discussion above

         * If pressed for more than 0.5 seconds, it changes the brightness of the light on the bound lighting bulb device (:ref:`light bulb <matter_light_bulb_sample>`).
           The brightness is changing from 0% to 100% with 1% increments every 300 milliseconds as long as **Button 2** is pressed.

>       Button 3:
>          Acts as User Active Mode Trigger button.
>          See :ref:`ug_matter_device_low_power_icd` for more information about Intermittently Connected Devices (ICD) and User Active Mode Trigger (UAT).

Reviewer 2: ```suggestion
      Button 3:
         Functions as the User Active Mode Trigger (UAT) button.
         For more information about Intermittently Connected Devices (ICD) and User Active Mode Trigger, see the :ref:`ug_matter_device_low_power_icd` documentation section. 
```

         .. note::

>             To activate this functionality, :ref:`activate the lit_icd snippet <matter_light_switch_snippets>`.

Reviewer 2: ```suggestion
            To enable this functionality, :ref:`activate the lit_icd snippet <matter_light_switch_snippets>`.
```

         * If pressed for more than 0.5 seconds, it changes the brightness of the light on the bound lighting bulb device (:ref:`light bulb <matter_light_bulb_sample>`).
           The brightness is changing from 0% to 100% with 1% increments every 300 milliseconds as long as **Button 1** is pressed.

>       Button 2:

Reviewer 2: Same suggestions as above
